File: test/single-language/python/uv/http-service/success-runtime-config/http_service_fixture/configuration.py
import os
import re
from dataclasses import dataclass


@dataclass(frozen=True)
class Configuration:
    port: int
    mode: str
    status: int
    label: str

    @classmethod
    def load(cls):
        raw = os.environ.get('PORT', '')
        if not re.fullmatch(r'[0-9]+', raw) or not 1 <= int(raw) <= 65535:
            raise ValueError('Invalid PORT')
        mode = 'config'
        label = os.environ.get('FIXTURE_LABEL', 'runtime-config-default') if mode == 'config' else 'deployment-smoke-ok'
        return cls(int(raw), mode, 200, label)

File: test/single-language/node/yarn/http-service/failure-configuration-rejected/src/configuration.js
function loadConfiguration() {
  const raw = process.env.PORT ?? "";
  const port = Number(raw);
  if (!/^[0-9]+$/.test(raw) || !Number.isInteger(port) || port < 1 || port > 65535) throw new Error("Invalid PORT");
  const mode = "smoke";
  const label = mode === "config" ? (process.env.FIXTURE_LABEL ?? "runtime-config-default") : "deployment-smoke-ok";
  return { port, mode, status: 200, label };
}
module.exports = { loadConfiguration };
